samples/sdl_joystick_event/main.c

```diff
@@ -0,0 +1,56 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/video.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+#define ARRAY_SIZE(x) (sizeof(x)/sizeof(x[0]))
+
+int main() {
+  XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_JOYSTICK);
+
+  int joystickIndex = 0;
+  SDL_Joystick *joy = SDL_JoystickOpen(joystickIndex);
+  if (joy == NULL) {
+    debugPrint("Could not open Joystick %i, exiting...", joystickIndex);
+    SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
+    Xsleep(2000);
+    return -1;
+  }
+
+  debugPrint("Number of axes   : %2i\n", SDL_JoystickNumAxes(joy));
+  debugPrint("Number of buttons: %2i\n", SDL_JoystickNumButtons(joy));
+  debugPrint("Number of balls  : %2i\n", SDL_JoystickNumBalls(joy));
+  debugPrint("Number of hats   : %2i\n", SDL_JoystickNumHats(joy));
+  debugPrint("\n");
+
+  while (true) {
+    SDL_Event event;
+    while (SDL_PollEvent(&event)) {
```

Comment:
Check if the joystick was disconnected, if it was, leave the loop, inform user / exit.

Reply:
Done. :+1:

Reply:
@dracc, you might want to update the other samples too. They're stuck in the loop even if the controller is disconnected.

Reply:
The GameController API doesn't provide such a check as far as I know, but the `sdl_gamecontroller_event` now uses it. :+1: